lib/room.rb

```diff
@@ -1,12 +1,25 @@
-class Rom
+class Room
+  attr_reader :category, :length, :width
 
   def initialize(category, length, width)
     @category = category
-    @length = 10
-    @width = width
+    @length = length
+    @width = width.to_i
   end
 
   def area
-    @length * @width
+    @length * @width.to_i
+  end
+
+
+  def paint
```

Comment:
The logic that you have for these paint methods looks like it is on the right track, but a better solution would look something like this:

```ruby
class Room
  ...
  def initialize(category, length, width)
    ...
    @is_painted = false
  end

  ...

  def paint
    @is_painted = true
  end

  def is_painted?
    @is_painted
  end
end
```